rest/admin_api.go

```diff
@@ -961,8 +981,10 @@ func (h *handler) handlePutCollectionConfigSync() error {
 			config := bucketDbConfig.Scopes[h.collection.ScopeName].Collections[h.collection.Name]
 			config.SyncFn = &js
 			bucketDbConfig.Scopes[h.collection.ScopeName].Collections[h.collection.Name] = config
+			updatedConfigStr = fmt.Sprintf(`{"scopes": {"%s": {"collections": { "%s": { "sync": "%s"}}}}}`, h.collection.ScopeName, h.collection.Name, js)
```

Comment:
This seems similar to the case discussed elsewhere - since the keyspace will already be included in the audit logging, I feel like we can omit the complexity of the nested structs for auditing purposes?  (same applies to other instances of this)

Reply:
The keyspace isn't typically included the audit logging unless we ask for it, and it would be an optional field for `base.AuditIDUpdateDatabaseConfig` since if you had an update from `/db/_config` you wouldn't have a keyspace.

Reply:
Based on that, it sounds like an optional `auditFieldKeyspace` on this event would be the right choice for the keyspaced endpoints that can modify config, as opposed to crafting an AuditFieldPayload property that doesn't match the request payload?  What do you think?

Reply:
I think conceptually from the user this is the action that they are doing as this config upsert and that is more

I guess we could have a freeform field:

"ks": "db.scope.collection"
"sync": nil <-- delete the custom sync function

and then make "ks" and "payload" optional? We can't log a payload for a DELETE operation and DELETE on a sync or import_filter doesn't reset to empty string, it resets to nil.